### ISSUE TITLE
[FR] Option to disable the serial interface

### ISSUE THREAD

**Issue by RudolphRiedel** [MarlinFirmware/Marlin#11952]
### Description

Is there an option to disable the serial interface completely?

### Steps to Reproduce

I would like to remove all the code for the serial interface from the build.
Just commenting-out "#define SERIAL_PORT 0" in Configuration.h does not work,
if not defined it will be set to "0" by MarlinSerial.h,

#### Additional Information

Not really an issue, more an option that does seem to be available for configuration.

I am using a CR-10 and an Ender-3 which both use a Melzi style board with a M1284p.
And even the example config for the CR-10 builds to 98.4% of program memory used.

Since I am printing from SD-card only I have no use at all for the serial interface.
Yet there is no option to remove it alltogether like you can disable SD-support.

**Comment by RudyBenner** [comment]
You could try -1 ?

**Comment by RudolphRiedel** [comment]
"-1" is not documented to be a valid option.
As well as with anything above 7 it will not compile.

**Comment by thinkyhead** [comment]
True, there is no way to remove the serial interface altogether, and it's a pretty major undertaking to add that option.

**Comment by RudolphRiedel** [comment]
Okay, got it, thank you.
And ignoring the fact that I am on project-overload already...
If I (or someone with more pressing need for this) would implement this option, would this be added to Marlin or not so likely?
Given of course it is done correctly, meets the guidelines for Marlin and has the desired effect.

**Comment by thinkyhead** [comment]
Such a change would …probably… be accepted. The one downside of having no serial interface is that we want to announce the firmware name and version and print a copyright/license message. We could add that as an extra boot screen, I suppose. (That’s mainly wanted to discourage GPL license violation.)

**Comment by ejtagle** [comment]
There are 2 ways to implement this: One of them would be to "stub out" all serial code (quite a huge thing!)... And the other would be to create an empty serial class with methods that do nothing (and that would be much, much easier). The compiler and the linker would do the rest (removing all dead code and unreferenced strings)
